Add Flowing Column block plugin

tommusrhodus opened the pull request:
Initial commit for the Flowing Column WordPress block plugin, including block registration, self-update functionality, configuration files, and source assets.

Fixes #47 

Enjoy @Nyiriland 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced Flowing Column block: configurable columns (1–6), minimum columns, unit-aware min width/gap/rule settings, editor controls, responsive stacking, and saved output via CSS variables.
  - Enabled plugin self-update / auto-update integration.

- **Documentation**
  - Added comprehensive Flowing Column readme and updated Scroll Progress Bar plugin metadata.

- **Chores**
  - Added EditorConfig, .gitignore, and package.json with build/dev scripts.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

tommusrhodus committed:
Add Flowing Column block plugin

Initial commit for the Flowing Column WordPress block plugin, including block registration, self-update functionality, configuration files, and source assets.

tommusrhodus commented:
@coderabbitai review

tommusrhodus committed:
Adds missing update URI

tommusrhodus committed:
Add column minimum width option to block

Introduces a new 'columnMinWidth' attribute to the flowing-column block, allowing users to set a minimum width for columns. Removes the 'stackOnMobile' option and related UI, and updates styles and logic to support the new minimum width setting.

tommusrhodus committed:
Remove layout settings from block.json

Deleted the 'layout' configuration from the block.json file for the flowing-column block, simplifying block settings and potentially relying on default layout behavior.

donnapep commented on flowing-column/readme.txt:
```diff
@@ -0,0 +1,55 @@
+=== Flowing Column ===
```
Would be good to update the ReadMe.

donnapep commented on flowing-column/src/flowing-column/block.json:
```diff
@@ -0,0 +1,84 @@
+	"name": "a8csp/flowing-column",
+	"version": "0.1.0",
+	"title": "Flowing Column",
+	"category": "widgets",
```
Probably makes more sense in `design` as that's where the column(s) block is.

donnapep commented on flowing-column/src/flowing-column/index.js:
```diff
@@ -0,0 +1,39 @@
+	/**
+	 * @see ./edit.js
+	 */
+	edit: Edit,
```
Nit: Seems easier to import this as `edit` instead of `Edit`, which will also match `save`.

tommusrhodus replied:
Good nitpick: Weirdly though this is how @wordpress/create-block sets this up 🤷‍♂️ 

I don't know why, but I assume good reasoning, for that I'll ignore on this one.

donnapep commented on flowing-column/src/flowing-column/save.js:
```diff
@@ -0,0 +1,48 @@
+		columnRuleColor,
+	} = attributes;
+
+	console.log( columnCount );
```
Oopsie!

donnapep commented on flowing-column/package.json:
```diff
@@ -0,0 +1,20 @@
+{
+	"name": "flowing-column",
+	"version": "0.1.0",
+	"description": "Example block scaffolded with Create Block tool.",
```
This has default content.

donnapep reviewed (approved):
Played around with it but not sure I understand how to control overflow if you can't set the height of the block. I left a few comments, though nothing critical.

tommusrhodus committed:
Add advanced column controls and responsive options

Enhances the Flowing Column block with new attributes for minimum columns, unit selection for column width, gap, and rule width, and improved responsive behavior. Updates UI controls in the editor, refines block.json schema, and expands documentation for better usability and flexibility in creating multi-column layouts.

tommusrhodus committed:
Improve column min width and rule defaults

Allows columnMinWidth to be set to 0 for auto width and updates help text in the editor. Ensures default values for column rule style and color are 'none' and '#000000' respectively, and improves type handling in save.js. Updates documentation to match new defaults.
